sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala

```diff
@@ -116,10 +116,20 @@ case class LogicalRDD(
       case e: Attribute => rewrite.getOrElse(e, e)
     }.asInstanceOf[SortOrder])
 
+    val rewrittenOriginLogicalPlan = originLogicalPlan.map { plan =>
+      assert(output == plan.output, "The output columns are expected to the same for output " +
```

Comment:
Actually I added this assertion on initialization as precondition, and realized canonicalization breaks the precondition. (output is canonicalized, but originLogicalPlan is not a target of canonicalization)

I wouldn't expect Spark calls newInstance against canonicalized node, but please correct me if I'm mistaken.

Reply:
I think Spark doesn't call newInstance with canonicalized node. cc @cloud-fan

Reply:
Yeah I can't think of any use case leveraging canonicalized node to even start with something.